optax/_src/transform.py

```diff
@@ -442,19 +447,32 @@ def init_fn(params):
     mu = optax.tree.zeros_like(params, dtype=mu_dtype)  # moment
     return ScaleByLionState(count=jnp.zeros([], jnp.int32), mu=mu)
 
+
+
   def update_fn(updates, state, params=None):
+
     del params
-    updates_new = jax.tree.map(
-        lambda g, m: jnp.sign((1.0 - b1) * g + b1 * m), updates, state.mu
-    )
+
+    def _comb(g, m):
+      x = (1.0 - b1) * g + b1 * m
+      if mode == "hard":
+        return jnp.sign(x)
+      elif mode == "smooth":
+        return jnp.tanh(smooth_beta * x)
+      elif mode == "refined":
+        # Keep small values linear, saturate to sign for large values.
+        return jnp.where(jnp.abs(x) < 1.0, x, jnp.sign(x))
+      else:
+        raise ValueError(f"Unknown mode {mode}")
```

Comment:
Add that this is for lion, e.g., `f'Unknown lion mode: {mode}. It needs to be one of ["hard", "smooth", "refined"].'`